### PR TITLE
Propogate transformed clipping regions to nested flows.

### LINKED ISSUE #10559: overflow:hidden clips too early (2)
Very similar to https://github.com/servo/servo/issues/10431:

```html
<style>
  div {
    position: absolute;
    height: 200px;
    width: 200px;
    top: 0;
  }
</style>

<div style="overflow: hidden; background: red;">
  <div style="transform: translateX(-100px);">
    <div style="right: -100px; background: green;">
      1 2 3 4 5 6 7 8 9 10 11 12 13
    </div>
  </div>
</div>
```

screen shot 2016-04-13 at 09 42 35

/cc @notriddle

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #10559

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10579)

### CODE REVIEW

File: components/layout/block.rs [hunk 1]
```diff
@@ -1965,7 +1965,48 @@ impl Flow for BlockFlow {
 
             flow::mut_base(kid).late_absolute_position_info =
                 late_absolute_position_info_for_children;
-            flow::mut_base(kid).clip = clip.clone();
+            let clip = if kid.class() == FlowClass::Block {
```
Comment (pcwalton) [inline components/layout/block.rs]: Why not use `flow.is_block_like()`? That'll catch tables and stuff too.

File: components/layout/display_list_builder.rs [hunk 2]
```diff
@@ -1743,46 +1743,13 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
...
-                            )
-                        };
-                        let off = Point2D::new(tx, ty);
-                        clip = clip.translate(&-off);
-                    }
-                    _ => {}
-                };
-            }
-        }
+        let tmp;
```
Comment (pcwalton) [inline components/layout/display_list_builder.rs]: I think I'd prefer a different variable name. `clipping_region` or something?

File: components/layout/block.rs (hunk 1 above)
Comment (notriddle) [inline components/layout/block.rs]: OK.

**Comment by notriddle** [comment]
Fixed both complaints.

**Comment by pcwalton** [comment]
@bors-servo: r+

Thanks!

**Comment by notriddle** [comment]
@bors-servo retry

* #10392

**Pull request merged**